### PR TITLE
Merge `toolNamesToPaths` and `toolNamesToTriples`

### PR DESCRIPTION
**Pull request by neonichu**
These started out as separate dictionaries, this merges them into one dictionary with a tuple value.

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test macOS

**Comment by neonichu** [comment]
@swift-ci please smoke test windows

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test windows

File: Sources/PackagePlugin/Context.swift [hunk 1]
```diff
@@ -64,12 +64,9 @@ public struct PluginContext {
...
     }
 
-    /// A mapping from tool names to their definitions. Not directly available
+    /// A mapping from tool names to their paths and triples. Not directly available
     /// to the plugin, but used by the `tool(named:)` API.
-    let toolNamesToPaths: [String: Path]
-
-    /// Supported triples per tool name; looked up in `tool(named:)`
-    let toolNamesToTriples: [String: [String]]
+    let accessibleTools: [String: (path: Path, triples: [String]?)]
```
Comment (abertelrud) [inline Sources/PackagePlugin/Context.swift:69]: Is there any semantic difference to having `.none` for triples vs `[]`?  In other words, should this really be optional or would just a plain array suffice?

**Review by abertelrud — COMMENTED** [review]
LGTM, just a question about whether there really is a semantic distinction between `.none` vs `[]` for triples and whether it couldn't be simplified to just an array (or does `.none` mean _any_, for example if the tool is a shell script?).

Thanks for doing this, I think it will make it a lot easier to extend since I suspect there will be more properties for tools in the future.

**Review by abertelrud — APPROVED** [review]

File: Sources/PackagePlugin/Context.swift (hunk 1 above)
Comment (abertelrud) [reply Sources/PackagePlugin/Context.swift:69]: Hm, I think I see:  `.none` means "any" (for example if the tool is actually a shell script that runs anywhere) and `[]` means "none" (i.e. that there are no compatible triples with the host).

Comment (neonichu) [reply Sources/PackagePlugin/Context.swift:69]: Yep, that's right. Basically for anything built from source, triples is `nil`, but for vended tools, there's always a value and `[]` means "no compatible triples".

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Pull request merged**